File: fix_syntax_errors.py
#!/usr/bin/env python3
import os
import re
import glob

# 要修复的目录
directories = [
    'app/api',
    'components',
    'lib',
    'types',
    'hooks',
    'utils',
    'constants',
    'contexts',
    'config'
]

# 修复模式
patterns = [
    {
        'pattern': re.compile(r'(\(\s*\w+:\s*\w+)\s*:\s*any\s*\)'),
        'replacement': r'\1)',
        'description': '修复箭头函数参数类型注解错误'
    },
    {
        'pattern': re.compile(r'(\w+\s*:\s*\w+)\s*:\s*any(\)|,)'),
        'replacement': r'\1\2',
        'description': '修复函数参数类型注解错误'
    }
]

def has_syntax_errors(content):
    return ':' in content and ': any' in content

def fix_file_syntax(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if not has_syntax_errors(content):
            return False
        
        fixed_content = content
        changes = []
        
        for pattern_info in patterns:
            before = fixed_content
            fixed_content = pattern_info['pattern'].sub(pattern_info['replacement'], fixed_content)
            
            if fixed_content != before:
                changes.append(pattern_info['description'])
        
        if fixed_content != content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(fixed_content)
            print(f"✅ 修复文件: {file_path}")
            for change in changes:
                print(f"   - {change}")
            return True
        
        return False
    except Exception as e:
        print(f"❌ 修复文件失败 {file_path}: {e}")
        return False

def fix_directory(dir_path):
    if not os.path.exists(dir_path):
        return 0
    
    fixed_count = 0
    
    for root, dirs, files in os.walk(dir_path):
        for file in files:
            if file.endswith(('.ts', '.tsx')):
                file_path = os.path.join(root, file)
                if fix_file_syntax(file_path):
                    fixed_count += 1
    
    return fixed_count

def main():
    print("🔧 开始修复TypeScript语法错误...\n")
    
    total_fixed = 0
    
    for dir_name in directories:
        print(f"检查目录: {dir_name}")
        fixed_count = fix_directory(dir_name)
        if fixed_count > 0:
            total_fixed += fixed_count
            print(f"  修复了 {fixed_count} 个文件\n")
        else:
            print("  没有发现需要修复的文件\n")
    
    print(f"🎉 修复完成！总共修复了 {total_fixed} 个文件")

if __name__ == "__main__":
    main()

File: fix-syntax-errors.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// 要修复的目录
const directories = [;
  'app/api',
  'components',
  'lib',
  'types',
  'hooks',
  'utils',
  'constants',
  'contexts',
  'config'
];

// 常见的语法错误模式
const errorPatterns = [;
  {
    // 修复箭头函数参数类型定义错误：(param: Type: any) -> (param: Type)
    pattern: /(\(\s*\w+:\s*\w+)\s*:\s*any\s*\)/g,
    replacement: '$1)',
    description: '修复箭头函数参数类型注解错误'
  },
  {
    // 修复reduce等函数的参数类型定义错误
    pattern: /(\w+\s*:\s*\w+)\s*:\s*any\s*(\)|,)/g,
    replacement: '$1$2',
    description: '修复函数参数类型注解错误'
  },
  {
    // 修复.map((param: Type) : any) 模式
    pattern: /(\(\s*\w+:\s*\w+\))\s*:\s*any/g,
    replacement: '$1',
    description: '修复.map箭头函数参数注解错误'
  },
  {
    // 修复.forEach(((param: Type) : any) 模式
    pattern: /(\(\s*\(\s*\w+:\s*\w+\)\s*:\s*any\s*\))/g,
    replacement: '($1)',
    description: '修复.forEach箭头函数参数注解错误'
  }
];

// 检查文件是否存在语法错误
function hasSyntaxErrors(content) {
  return /:\s*\w+\s*:\s*any/.test(content);
}

// 修复文件中的语法错误
function fixFileSyntax(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let fixed = false;
    let changes = [];

    errorPatterns.forEach((patternInfo, index) => {
      const before = content;
      content = content.replace(patternInfo.pattern, patternInfo.replacement);
      
      if (content !== before) {
        fixed = true;
        changes.push(patternInfo.description);
      }
    });

    if (fixed) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`✅ 修复文件: ${filePath}`);
      changes.forEach(change => console.log(`   - ${change}`));
      return true;
    }

    return false;
  }
  } catch (error) {
    console.error(`❌ 修复文件失败 ${filePath}:`, error.message);
    return false;
  }
}

// 遍历目录并修复文件
function fixDirectory(dirPath) {
  if (!fs.existsSync(dirPath)) {
    return 0;
  }

  let fixedCount = 0;
  const files = fs.readdirSync(dirPath);

  files.forEach(file => {
    const fullPath = path.join(dirPath, file);
    const stat = fs.statSync(fullPath);

    if (stat.isDirectory()) {
      fixedCount += fixDirectory(fullPath);
    } else if (stat.isFile() && (file.endsWith('.ts') || file.endsWith('.tsx'))) {
      if (hasSyntaxErrors(fs.readFileSync(fullPath, 'utf8'))) {
        if (fixFileSyntax(fullPath)) {
          fixedCount++;
        }
      }
    }
  });

  return fixedCount;
}

// 主函数
function main() {
  console.log('🔧 开始修复TypeScript语法错误...\n');

  let totalFixed = 0;

  directories.forEach(dir => {
    console.log(`检查目录: ${dir}`);
    const fixedCount = fixDirectory(dir);
    if (fixedCount > 0) {
      totalFixed += fixedCount;
      console.log(`  修复了 ${fixedCount} 个文件\n`);
    } else {
      console.log('  没有发现需要修复的文件\n');
    }
  });

  console.log(`🎉 修复完成！总共修复了 $ 个文件`);
}

// 运行
main();